server/monitor/src/main/java/org/apache/accumulo/monitor/rest/tservers/TabletServerResource.java

```diff
@@ -0,0 +1,336 @@
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response.Status;
+
+import org.apache.accumulo.core.Constants;
+import org.apache.accumulo.core.client.impl.ClientContext;
+import org.apache.accumulo.core.client.impl.Tables;
+import org.apache.accumulo.core.conf.Property;
+import org.apache.accumulo.core.data.impl.KeyExtent;
+import org.apache.accumulo.core.master.thrift.MasterMonitorInfo;
+import org.apache.accumulo.core.master.thrift.RecoveryStatus;
+import org.apache.accumulo.core.master.thrift.TabletServerStatus;
+import org.apache.accumulo.core.rpc.ThriftUtil;
+import org.apache.accumulo.core.tabletserver.thrift.ActionStats;
+import org.apache.accumulo.core.tabletserver.thrift.TabletClientService;
+import org.apache.accumulo.core.tabletserver.thrift.TabletStats;
+import org.apache.accumulo.core.trace.Tracer;
+import org.apache.accumulo.core.util.AddressUtil;
+import org.apache.accumulo.core.zookeeper.ZooUtil;
+import org.apache.accumulo.monitor.Monitor;
+import org.apache.accumulo.monitor.rest.master.MasterResource;
+import org.apache.accumulo.server.client.HdfsZooInstance;
+import org.apache.accumulo.server.master.state.DeadServerList;
+import org.apache.accumulo.server.util.ActionStatsUpdator;
+
+import com.google.common.net.HostAndPort;
+
+/**
+ *
+ * Generates tserver lists as JSON objects
+ *
+ * @since 2.0.0
+ *
+ */
+@Path("/tservers")
+@Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML})
+public class TabletServerResource {
+
+  // Variable names become JSON keys
+  private TabletStats total, historical;
+
+  /**
+   * Generates tserver summary
+   *
+   * @return tserver summary
+   */
+  @GET
+  public TabletServers getTserverSummary() {
+    MasterMonitorInfo mmi = Monitor.getMmi();
+    if (null == mmi) {
+      throw new WebApplicationException(Status.INTERNAL_SERVER_ERROR);
+    }
+
+    TabletServers tserverInfo = new TabletServers(mmi.tServerInfo.size());
+    for (TabletServerStatus status : mmi.tServerInfo) {
+      tserverInfo.addTablet(new TabletServer(status));
+    }
+
+    tserverInfo.addBadTabletServer(new MasterResource().getTables());
+
+    return tserverInfo;
+  }
+
+  /**
+   * REST call to clear dead servers from list
+   *
+   * @param server
+   *          Dead server to clear
+   */
+  @POST
+  @Consumes(MediaType.TEXT_PLAIN)
+  public void clearDeadServer(@QueryParam("server") String server) throws Exception {
+    DeadServerList obit = new DeadServerList(ZooUtil.getRoot(Monitor.getContext().getInstance()) + Constants.ZDEADTSERVERS);
+    obit.delete(server);
+  }
+
+  /**
+   * Generates a recovery tserver list
+   *
+   * @return Recovery tserver list
+   */
+  @Path("recovery")
+  @GET
+  public Map<String,List<Map<String,String>>> getTserverRecovery() {
```

Comment:
Yes, it has to return a maps of maps (in this case), since the first map uses the Key for JSON (this is another way of writing what I did with classes, I might change this to have another class, but I used this instead since it was faster to code (but can get confusing)